modules/checklist/steps-manager.php

```diff
@@ -32,12 +33,13 @@ public function get_steps_for_frontend() : array {
 
 		foreach ( self::$step_ids as $step_id ) {
 			$instance = $this->step_instances[ $step_id ];
-			$is_marked_as_completed = $instance->is_marked_as_completed();
+			$instance->maybe_immutably_mark_as_completed();
```

Comment:
Whenever the checklist loads on the frontend, there needs to be a check to see if steps were completed between the last time it was loaded